main_test.go

```diff
@@ -194,3 +203,53 @@ func registeredTestWslInstances() ([]wsl.Distro, error) {
 
 	return distros, nil
 }
+
+// defaultDistro gets the default distro's name via wsl.exe to bypass wsl.DefaultDistro in order to
+// better decouple tests.
+func defaultDistro() (string, error) {
+	out, err := exec.Command("powershell.exe", "-Command", "$env:WSL_UTF8=1; wsl.exe --list --verbose").CombinedOutput()
+	if strings.Contains(string(out), "Windows Subsystem for Linux has no installed distributions.") {
+		return "", nil
+	}
+	if err != nil {
+		return "", fmt.Errorf("failed to find current default distro: %v", err)
+	}
+	s := bufio.NewScanner(bytes.NewReader(out))
+	s.Scan() // Ignore first line (table header)
+	for s.Scan() {
+		line := s.Text()
+		if !strings.HasPrefix(line, "*") {
+			continue
+		}
+		data := strings.Fields(line)
+		if len(data) < 2 {
+			return "", fmt.Errorf("failed to parse 'wsl.exe --list --verbose' output, line %q", line)
+		}
+		return data[1], nil
+	}
+
+	if err := s.Err(); err != nil {
+		return "", err
+	}
+
+	return "", fmt.Errorf("failed to find default distro in 'wsl.exe --list --verbose' output:\n%s", string(out))
+}
+
+// backUpDefaultDistro returns a function to restore the default distro so the machine is restored to
+// its pre-testing state.
+func backUpDefaultDistro() (func(), error) {
+	distro, err := defaultDistro()
+	if err != nil {
+		return nil, fmt.Errorf("failed to back up default distro: %v", err)
+	}
+	if len(distro) == 0 {
+		return func() {}, nil // No distros registered: no backup needed
+	}
+	restore := func() {
+		out, err := exec.Command("wsl.exe", "--set-default", distro).CombinedOutput() //nolint: gosec
```

Comment:
ditto on gosec nolint.

Reply:
I think you can prefix the paragraph then with the nolint gosec? Something like this: https://github.com/ubuntu/adsys/blob/main/internal/policies/privilege/privilege.go#L92.

This way. the comment and the notlint directive are in the same paragraph and are connected.

Reply:
Oh this looks much nicer. Done 👍.